### PR TITLE
Prevent Handler from adding invalid searches to Watchlist

### PR DESCRIPTION
**Pull request by SHTDJ**
Fixes #
Entry is only added to watchlist if valid
## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: client/App.jsx [hunk 1]
```diff
@@ -120,8 +120,9 @@ class App extends Component {
     if (this.props.watch.length && this.props.watch[0] === term) {
       return;
     }
-
+if (isTX(term) && isBlock(term) && isAddress(term)) {
```
Comment (dustinengle) [inline client/App.jsx]: I think we need OR statements here or the check will require that the term is a tx, block, and address.
```
if (isTX(term) || isBlock(term) || isAddress(term)) {
```

File: lib/blockchain.js [hunk 2]
```diff
@@ -313,7 +313,7 @@ const isBlock = (s) => {
 };
 
 const isTX = (s) => {
-  return typeof(s) === 'string';
+  return s.lenght === 64 && s.substr(0, 4) != '0000';
```
Comment (dustinengle) [inline lib/blockchain.js]: ```length```

**Comment by SHTDJ** [comment]
yeah error on my part sry

**Comment by SHTDJ** [comment]
Should be in line now

**Pull request merged**